From f7f8676db491fee877020ad654fe46f0099e7274 Mon Sep 17 00:00:00 2001
From: Mark Harmstone <mark@harmstone.com>
Date: Tue, 3 Nov 2015 16:22:09 -0600
Subject: winepulse: add stub for GetPropValue

Edited by Maarten Lankhorst: No support for multiple devices in winepulse yet.

Signed-off-by: Andrew Eikum <aeikum@codeweavers.com>
---
 dlls/winepulse.drv/mmdevdrv.c         | 7 +++++++
 dlls/winepulse.drv/winepulse.drv.spec | 1 +
 2 files changed, 8 insertions(+)

diff --git a/dlls/winepulse.drv/mmdevdrv.c b/dlls/winepulse.drv/mmdevdrv.c
index e5ba869..64e6c2a 100644
--- a/dlls/winepulse.drv/mmdevdrv.c
+++ b/dlls/winepulse.drv/mmdevdrv.c
@@ -50,6 +50,7 @@
 #include "initguid.h"
 #include "ks.h"
 #include "ksmedia.h"
+#include "propkey.h"
 #include "mmdeviceapi.h"
 #include "audioclient.h"
 #include "endpointvolume.h"
@@ -3198,3 +3199,9 @@ HRESULT WINAPI AUDDRV_GetAudioSessionManager(IMMDevice *device,
     *out = &This->IAudioSessionManager2_iface;
     return S_OK;
 }
+
+HRESULT WINAPI AUDDRV_GetPropValue(GUID *guid, const PROPERTYKEY *prop, PROPVARIANT *out)
+{
+    TRACE("%s, (%s,%u), %p\n", wine_dbgstr_guid(guid), wine_dbgstr_guid(&prop->fmtid), prop->pid, out);
+    return E_NOTIMPL;
+}
diff --git a/dlls/winepulse.drv/winepulse.drv.spec b/dlls/winepulse.drv/winepulse.drv.spec
index 288de87..7aeb175 100644
--- a/dlls/winepulse.drv/winepulse.drv.spec
+++ b/dlls/winepulse.drv/winepulse.drv.spec
@@ -3,6 +3,7 @@
 @ stdcall -private GetEndpointIDs(long ptr ptr ptr ptr) AUDDRV_GetEndpointIDs
 @ stdcall -private GetAudioEndpoint(ptr ptr ptr) AUDDRV_GetAudioEndpoint
 @ stdcall -private GetAudioSessionManager(ptr ptr) AUDDRV_GetAudioSessionManager
+@ stdcall -private GetPropValue(ptr ptr ptr) AUDDRV_GetPropValue
 
 # WinMM driver functions
 @ stdcall -private DriverProc(long long long long long) winealsa.drv.DriverProc
-- 
2.6.2

